backend/src/tasks/test/dnstwist.test.ts

```diff
@@ -0,0 +1,163 @@
+import { handler as dnstwist } from '../dnstwist';
+import {
+  connectToDatabase,
+  Organization,
+  Domain,
+  Scan,
+  Vulnerability
+} from '../../models';
+import { spawnSync } from 'child_process';
+
+const RealDate = Date;
+
+jest.mock('child_process', () => ({
+  spawnSync: jest.fn()
+}));
+
+describe('dnstwist', () => {
+  let scan;
+  let organization;
+  const domains: Domain[] = [];
+  beforeAll(async () => {
+    (spawnSync as jest.Mock).mockImplementation(() => ({
+      status: 0,
+      signal: null,
+      output: [null, '', ''],
+      pid: 3,
+      stdout: `[
+                {
+                    "fuzzer": "Homoglyph",
+                    "domain-name": "test-domain.one",
+                    "dns-a": ["21.22.23.24"]
+                },
+                {
+                    "fuzzer": "Original",
+                    "domain-name": "test-domain.two",
+                    "dns-a": ["01.02.03.04"],
+                    "dns-mx": ["localhost"]
+                },
+                {
+                    "fuzzer": "tls",
+                    "domain-name": "test-domain.three",
+                    "dns-a": ["10.11.12.13"],
+                    "dns-ns": ["example.link"]
+                }
+            ]`,
+      stderr: ''
+    }));
+  });
+  beforeEach(async () => {
+    await connectToDatabase();
+    global.Date.now = jest.fn(() => new Date('2019-04-22T10:20:30Z').getTime());
+    organization = await Organization.create({
+      name: 'test-' + Math.random(),
+      rootDomains: ['test-' + Math.random()],
+      ipBlocks: [],
+      isPassive: false
+    }).save();
+    scan = await Scan.create({
+      name: 'dnstwist',
+      arguments: {},
+      frequency: 999
+    }).save();
+    global.Date.now = jest.fn(() => new Date('2019-04-22T10:20:30Z').getTime());
+    jest.mock('../helpers/getIps', () => domains);
+  });
+
+  afterEach(() => {
+    global.Date = RealDate;
+    jest.unmock('../helpers/getIps');
+  });
+
+  const checkDomains = async (organization) => {
+    const domains = await Domain.find({
+      where: { organization },
+      relations: ['organization', 'services']
+    });
+    expect(
+      domains
+        .sort((a, b) => a.name.localeCompare(b.name))
+        .map((e) => ({
+          ...e,
+          services: e.services.map((s) => ({
+            ...s,
+            id: null,
+            updatedAt: null,
+            createdAt: null
+          })),
+          organization: null,
+          id: null,
+          updatedAt: null,
+          createdAt: null,
+          syncedAt: null,
+          name: null
+        }))
+    ).toMatchSnapshot();
+    expect(domains.filter((e) => !e.organization).length).toEqual(0);
+  };
+
+  test('basic test', async () => {
+    await dnstwist({
+      organizationId: organization.id,
+      organizationName: 'organizationName',
+      scanId: scan.id,
+      scanName: 'scanName',
+      scanTaskId: 'scanTaskId'
+    });
+
+    await checkDomains(organization);
```

Comment:
Given that the domains just gives an empty list (`[]`) (see the snapshot), and the fact that we really want to check for created vulnerabilities, not domains, let's simplify this by removing the `checkDomains` function entirely and just running:

```suggestion
    const vulns = await Vulnerability.find({
      domain: domain
    });
    expect(vulns.length).toEqual(0);
```

Reply:
@aloftus23 I think you removed checkDomains, but you forgot to add the Vulnerability.find check. Would you be able to add this check in another PR?

Reply:
Actually I had it 104-107. Is it not running properly? @epicfaace